src/_arraykit.c

```diff
@@ -198,11 +198,12 @@ AK_ArrayDeepCopy(PyArrayObject *array, PyObject *memo)
                 array,
                 dtype,
                 NPY_ARRAY_ENSURECOPY);
-        if (memo) {
-            if (!array_new || PyDict_SetItem(memo, id, array_new)) {
-                Py_XDECREF(array_new);
-                goto error;
-            }
+        if (!array_new) { // we unnecessarily incref'ed the dtype
```

Comment:
This comment no longer applies:

```suggestion
        if (!array_new) {
```